src/libaktualizr/http/httpclient.cc

```diff
@@ -226,11 +236,16 @@ HttpResponse HttpClient::download(const std::string& url, curl_write_callback ca
   curlEasySetoptWrapper(curl_download, CURLOPT_LOW_SPEED_LIMIT, speed_limit_bytes_per_sec_);
   curlEasySetoptWrapper(curl_download, CURLOPT_RESUME_FROM, from);
 
-  CURLcode result = curl_easy_perform(curl_download);
-  curl_easy_getinfo(curl_download, CURLINFO_RESPONSE_CODE, &http_code);
-  HttpResponse response("", http_code, result, (result != CURLE_OK) ? curl_easy_strerror(result) : "");
-  curl_easy_cleanup(curl_download);
-  return response;
+  auto promise = std::make_shared<std::promise<HttpResponse>>();
+  std::thread([curlp, promise]() {
+    CURLcode result = curl_easy_perform(curlp.get());
+    long http_code;
+    curl_easy_getinfo(curlp.get(), CURLINFO_RESPONSE_CODE, &http_code);
+    HttpResponse response("", http_code, result, (result != CURLE_OK) ? curl_easy_strerror(result) : "");
+    promise->set_value(response);
+  })
+      .detach();
+  return promise->get_future();
```

Comment:
It's more like a style preference, but wouldn't it be cleaner to get future first and then move the promise?

Reply:
Hm... I don't know, it's copied not moved after all. Does it look confusing?

Reply:
Yes, a bit. The whole point of creating a shared pointer is only to do promise->get_future() after the lambda statement.

Reply:
Got it, sorry. Fixed.